.rhdh/scripts/install.sh

```diff
@@ -53,15 +55,28 @@ echo "Using ${CHART_URL} to install Helm chart"
 # choose namespace for the install (or create if non-existant)
 oc new-project "$namespace" || oc project "$namespace"
 
-# TODO: RHIDP-6668 generate rhdh-next-ci-repo.yaml while installing so we don't have to publish a new file every time
-# TODO: RHIDP-6668 publish an index.yaml with every tarball pushed to quay.io/rhdh/chart; save them in rhdh-chart repo (one per CI versioned branch)
-# if [[ "$CV" == *"-CI" ]] || [[ $chartrepo -eq 1 ]]; then
-# see samples at 
-# https://github.com/rhdh-bot/openshift-helm-charts/blob/rhdh-1-rhel-9/installation/index.yaml#L19
-# https://github.com/rhdh-bot/openshift-helm-charts/blob/rhdh-1-rhel-9/installation/index.yaml#L49-L50
-# https://github.com/rhdh-bot/openshift-helm-charts/blob/rhdh-1-rhel-9/installation/rhdh-next-ci-repo.yaml#L8
-#     oc apply -f https://github.com/redhat-developer/rhdh-chart/raw/redhat-developer-hub-"${CV}"/installation/rhdh-next-ci-repo.yaml
-# fi
+# generate repo.yaml and index.yaml so we don't have to publish a new file every time
+if [[ "$CV" == *"-CI" ]] || [[ $chartrepo -eq 1 ]]; then
+  mkdir -p /tmp/"$CV"-unpacked && pushd /tmp/"$CV"-unpacked >/dev/null 2>&1 || exit 1
+  helm pull oci://quay.io/rhdh/chart --version "$CV" -d /tmp/"$CV"-unpacked # get tarball
+  helm repo index /tmp/"$CV"-unpacked # create index.yaml
+  cat <<EOF > repo.yaml
+apiVersion: helm.openshift.io/v1beta1
+kind: HelmChartRepository
+metadata:
+  name: rhdh-next-ci-repo
+spec:
+  connectionConfig:
+    file: >-
+      ./index.yaml
```

Comment:
Would this actually work? I couldn't see such `file` property in the `connectionConfig` spec: https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/config_apis/helmchartrepository-helm-openshift-io-v1beta1#spec-connectionconfig
And I tested it in an OCP cluster, and this is what I am seeing with this chart repo created:

![image](https://github.com/user-attachments/assets/a5ffbc7f-b643-4908-aba2-fbb4d164092f)

![image](https://github.com/user-attachments/assets/ef70d32e-58bc-4ccc-a079-e1eaff305fc8)

Reply:
I had the correct version appear here: https://issues.redhat.com/secure/attachment/13414165/screenshot-1.png 

Perhaps you're running an older (cached) version of the chart?

Reply:
as discussed in slack... afer manually pushing locally, this works:

```
 ./scripts/install.sh 1.7-51-CI -n rhdh-ci
```

but this has version 4.2.1 in the helm chart version picker:

```
 ./scripts/install.sh 1.7-47-CI -n rhdh-ci
```

so... applied another fix to the publishing script to replace `backstage` with `redhat-developer-hub` when doing the `all` mode push; building in https://konflux-ui.apps.stone-prod-p02.hjvn.p1.openshiftapps.com/ns/rhdh-tenant/applications/rhdh-1/pipelineruns/rhdh-hub-1-on-push-s5wcm 

if that one completes OK I cn try installing in on ROSA 4.18 and see what happens in the UI

Reply:
Armel said:

> It should be good to merge your PR then once it works with 1.7-52-CI. Thanks.

Reply:
![image](https://github.com/user-attachments/assets/c4f4fed8-d4ca-4f6f-95ee-8e366ba707c4)